lib/code_ownership/private/ownership_mappers/directory_ownership.rb

```diff
@@ -74,36 +74,47 @@ def owner_for_codeowners_file(codeowners_file)
           )
         end
 
-        # takes a file and finds the relevant `.codeowner` file by walking up the directory
+        # Takes a file and finds the relevant `.codeowner` file by walking up the directory
         # structure. Example, given `a/b/c.rb`, this looks for `a/b/.codeowner`, `a/.codeowner`,
         # and `.codeowner` in that order, stopping at the first file to actually exist.
-        # We do additional caching so that we don't have to check for file existence every time
+        # If the parovided file is a directory, it will look for `.codeowner` in that directory and then upwards.
+        # We do additional caching so that we don't have to check for file existence every time.
         sig { params(file: String).returns(T.nilable(CodeTeams::Team)) }
         def map_file_to_relevant_owner(file)
           file_path = Pathname.new(file)
-          path_components = file_path.each_filename.to_a.map { |path| Pathname.new(path) }
+          team = T.let(nil, T.nilable(CodeTeams::Team))
 
-          (path_components.length - 1).downto(0).each do |i|
-            potential_relative_path_name = T.must(path_components[0...i]).reduce(Pathname.new('')) { |built_path, path| built_path.join(path) }
-            potential_codeowners_file = potential_relative_path_name.join(CODEOWNERS_DIRECTORY_FILE_NAME)
+          if File.directory?(file)
+            team = get_team_from_codeowners_file_within_directory(file_path)
+          end
+
+          # Pathname of '.' and '/' are the root for relative and absolute paths respectively
+          while team.nil? && file_path != Pathname('.') && file_path != Pathname('/')
```

Comment:
I'm not sure how yet, but it would be good to have some specs around the logic to keep it from ascending beyond the relative or absolute root.

Reply:
The existing specs test this.

Reply:
I added an absolute test to the spec that I added as well. But I did also confirm that if this loop is not working, other tests will fail with the newly added one.